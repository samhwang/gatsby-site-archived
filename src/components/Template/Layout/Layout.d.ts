type SectionType = {
  id: string;
  content: React.ReactNode | Element;
};

type LayoutProps = {
  sections: SectionType[];
};

type LayoutRenderProps = LayoutProps & {
  siteMetadata: SiteMetadata;
};
