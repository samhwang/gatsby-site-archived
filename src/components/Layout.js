import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { StaticQuery, graphql } from 'gatsby';

import '../assets/sass/resume.scss';

function Layout(props) {
  const { children } = props;
  return (
    <StaticQuery
      query={graphql`
          query SiteTitleQuery {
            site {
              siteMetadata {
                title
              }
            }
          }
        `}
      render={data => (
        <>
          <Helmet
            title={data.site.siteMetadata.title}
            meta={[
              { name: 'description', content: 'Resume' },
            ]}
          >
            <html lang="en" />
          </Helmet>
          <div className={'main-body'}>{children}</div>
        </>
      )}
    />
  );
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
