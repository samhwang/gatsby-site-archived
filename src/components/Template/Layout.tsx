import React from 'react';
import Helmet from 'react-helmet';
import { StaticQuery, graphql } from 'gatsby';
import Sidebar from './Sidebar';
import Footer from './Footer';
import '../../assets/sass/resume.scss';

function Layout(props: { children: React.ReactNode }) {
  const { children } = props;
  return (
    <StaticQuery
      query={graphql`
        query SiteTitleQuery {
          site {
            siteMetadata {
              title
            }
          }
        }
      `}
      render={data => (
        <>
          <Helmet
            title={data.site.siteMetadata.title}
            meta={[{ name: 'description', content: 'Resume' }]}
          >
            <html lang="en" />
          </Helmet>
          <div className="main-body">
            <Sidebar />
            <div className="container-fluid p-0">
              {children}
              <Footer />
            </div>
          </div>
        </>
      )}
    />
  );
}

export default Layout;
