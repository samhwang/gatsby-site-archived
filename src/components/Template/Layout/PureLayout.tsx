import React, { Fragment } from 'react';
import { Helmet } from 'react-helmet';
import Sidebar from '../Sidebar';
import Footer from '../Footer';
import '../../../assets/sass/resume.scss';
import '../../Icons';

function PureLayout({ sections, siteMetadata }: LayoutRenderProps) {
  const { title, description } = siteMetadata;

  const children = sections.map(({ id, content }: SectionType) => (
    <Fragment key={id}>
      {content}
      <hr className="m-0" />
    </Fragment>
  ));

  return (
    <>
      <Helmet>
        <title>{title}</title>
        <meta name="description" content={description} />
        <meta
          name="viewport"
          content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no"
        />
        <html lang="en" />
      </Helmet>
      <div className="main-body">
        <Sidebar />
        <div className="container-fluid p-0">{children}</div>
        <Footer />
      </div>
    </>
  );
}

export default PureLayout;
