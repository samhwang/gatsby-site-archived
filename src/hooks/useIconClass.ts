import { useMemo, useCallback } from 'react';
import { useToggle } from 'rooks';

const useIconClass = (name: string) => {
  const DEFAULT_ICON_CLASS = useMemo<string>(() => `devicon-${name}`, []);
  const toggleColorIcon = useCallback(
    (currentClass) =>
      currentClass === DEFAULT_ICON_CLASS
        ? `${DEFAULT_ICON_CLASS} colored`
        : DEFAULT_ICON_CLASS,
    []
  );

  const [currentIconClass, toggleIconClass] = useToggle(
    DEFAULT_ICON_CLASS,
    toggleColorIcon
  );

  return [currentIconClass, toggleIconClass] as const;
};

export default useIconClass;
