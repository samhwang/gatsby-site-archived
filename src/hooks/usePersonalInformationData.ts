import { useStaticQuery, graphql } from 'gatsby';

interface PersonalInformation {
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  socialLinks: {
    icon: string;
    name: string;
    url: string;
  }[];
  projects: {
    title: string;
    description: string;
    url?: string;
  }[];
  education: {
    institute: string;
    degree: string;
    major: string;
    duration: string;
  }[];
  experience: {
    title: string;
    companyName: string;
    description: string[];
    duration: string;
    techIcons: string[];
    technologies: string[];
  }[];
  skills: {
    title: string;
    description: string[];
  }[];
  techStack: string[];
}

const usePersonalInformationData = (): PersonalInformation => {
  const PersonalInfoQuery = graphql`
    query PersonalInfoQuery {
      site {
        siteMetadata {
          personalInformation {
            firstName
            lastName
            address
            email
            socialLinks {
              icon
              name
              url
            }
            projects {
              title
              description
              url
            }
            education {
              institute
              degree
              major
              duration
            }
            experience {
              title
              companyName
              description
              duration
              techIcons
              technologies
            }
            skills {
              title
              description
            }
            techStack
          }
        }
      }
    }
  `;

  const data = useStaticQuery(PersonalInfoQuery);
  const { personalInformation } = data.site.siteMetadata;

  return personalInformation;
};

export default usePersonalInformationData;
