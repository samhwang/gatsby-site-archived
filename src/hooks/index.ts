import usePersonalInformationData from './usePersonalInformationData';
import useSiteMetadata from './useSiteMetadata';
import useIconClass from './useIconClass';

export { useSiteMetadata, usePersonalInformationData, useIconClass };
