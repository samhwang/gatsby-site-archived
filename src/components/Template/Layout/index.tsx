import type { FC } from 'react';
import { CssBaseline, Container, Grid } from '@material-ui/core';
import SEO from '../SEO';
import Footer from '../Footer';
import '../../Icons';
import { useSiteMetadata } from '../../../hooks';

export type LayoutProps = {
  title: string;
  description: string;
};

const Layout: FC<LayoutProps> = ({ children, title, description }) => {
  const defaultFallback = useSiteMetadata();

  return (
    <>
      <CssBaseline />
      <Container maxWidth="md">
        <SEO
          title={title}
          description={description}
          defaultFallback={defaultFallback}
        />
        <Grid container>
          {children}
          <Footer />
        </Grid>
      </Container>
    </>
  );
};

export default Layout;
