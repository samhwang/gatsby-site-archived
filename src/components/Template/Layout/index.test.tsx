import { render } from '@testing-library/react';
import { mocked } from 'ts-jest/utils';
import Layout from '.';
import { useSiteMetadata, usePersonalInformationData } from '../../../hooks';

jest.mock('../../../hooks');
const mockMetadataHook = mocked(useSiteMetadata);
const mockDataHook = mocked(usePersonalInformationData);

describe('Layout rendering', () => {
  it('Should match snapshot', () => {
    const mockMetadata = {
      title: 'Sam Huynh',
      description: 'Online profile of Sam Huynh.',
    };
    mockMetadataHook.mockReturnValueOnce(mockMetadata);

    const mockData: any = {
      firstName: 'Sam',
      lastName: 'Huynh',
    };
    mockDataHook.mockReturnValueOnce(mockData);

    const LayoutComponent = (
      <Layout title="Test Render" description="Test Render">
        <h1>This is a test!</h1>
      </Layout>
    );
    const tree = render(LayoutComponent);
    expect(tree).toMatchSnapshot();
  });
});
