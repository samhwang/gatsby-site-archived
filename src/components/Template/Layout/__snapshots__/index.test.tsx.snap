// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout rendering Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
            >
              Sam Huynh
            </h6>
            <div
              class="MuiGrid-root makeStyles-emptySpace-6"
            />
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              <a
                class="makeStyles-link-5"
                href="/"
              >
                Home
              </a>
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              <a
                class="makeStyles-link-5"
                href="/blog"
              >
                Blog
              </a>
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              <a
                class="makeStyles-link-5"
                href="/resume"
              >
                Resume
              </a>
            </h6>
            
          </div>
        </header>
        <main
          class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthMd"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
          >
            <h1>
              This is a test!
            </h1>
          </div>
        </main>
        <footer
          class="makeStyles-footer-7"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthMd"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
            >
              © 2021 
              <a
                href="https://github.com/samhwang"
              >
                Sam Huynh
              </a>
              . Built with 
              <a
                href="https://www.gatsbyjs.org/"
              >
                GatsbyJS
              </a>
               and hosted on 
              <a
                href="https://pages.github.com/"
              >
                GitHub Pages.
              </a>
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
          >
            Sam Huynh
          </h6>
          <div
            class="MuiGrid-root makeStyles-emptySpace-6"
          />
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            <a
              class="makeStyles-link-5"
              href="/"
            >
              Home
            </a>
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            <a
              class="makeStyles-link-5"
              href="/blog"
            >
              Blog
            </a>
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            <a
              class="makeStyles-link-5"
              href="/resume"
            >
              Resume
            </a>
          </h6>
          
        </div>
      </header>
      <main
        class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthMd"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
        >
          <h1>
            This is a test!
          </h1>
        </div>
      </main>
      <footer
        class="makeStyles-footer-7"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
          >
            © 2021 
            <a
              href="https://github.com/samhwang"
            >
              Sam Huynh
            </a>
            . Built with 
            <a
              href="https://www.gatsbyjs.org/"
            >
              GatsbyJS
            </a>
             and hosted on 
            <a
              href="https://pages.github.com/"
            >
              GitHub Pages.
            </a>
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
